src/SprykerEco/Zed/PunchoutCatalogs/Communication/PunchoutCatalogsCommunicationFactory.php

```diff
@@ -0,0 +1,185 @@
+     *
+     * @return \Symfony\Component\Form\FormInterface
+     */
+    public function getPunchoutCatalogConnectionForm(?PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer = null): FormInterface
+    {
+        $idPunchoutCatalogConnectionFormDataProvider = $this->createPunchoutCatalogConnectionFormDataProvider();
+
+        return $this->getFormFactory()
+            ->create(
+                PunchoutCatalogConnectionForm::class,
+                $idPunchoutCatalogConnectionFormDataProvider->getData($punchoutCatalogConnectionTransfer),
+                $idPunchoutCatalogConnectionFormDataProvider->getOptions()
+            );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\PunchoutCatalogs\Communication\Form\DataProvider\PunchoutCatalogConnectionFormDataProvider
+     */
+    public function createPunchoutCatalogConnectionFormDataProvider(): PunchoutCatalogConnectionFormDataProvider
+    {
+        return new PunchoutCatalogConnectionFormDataProvider(
+            $this->getCompanyBusinessUnitFacade(),
+            $this->getPunchoutCatalogConnectionFormatPlugins(),
+            $this->getPunchoutCatalogConnectionTypePlugins()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\PunchoutCatalogs\Communication\Form\DataProvider\PunchoutCatalogSetupRequestConnectionTypeFormDataProvider
+     */
+    public function createPunchoutCatalogSetupRequestConnectionTypeFormDataProvider(): PunchoutCatalogSetupRequestConnectionTypeFormDataProvider
+    {
+        return new PunchoutCatalogSetupRequestConnectionTypeFormDataProvider(
+            $this->getCompanyBusinessUnitFacade(),
+            $this->getCompanyUserFacade()
+        );
+    }
+
+    /**
+     * @return \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionQuery
+     */
+    public function getPunchoutCatalogConnectionPropelQuery(): PgwPunchoutCatalogConnectionQuery
+    {
+        return $this->getProvidedDependency(PunchoutCatalogsDependencyProvider::PROPEL_QUERY_PUNCHOUT_CATALOG_CONNECTION);
+    }
+
+    /**
+     * @return \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogTransactionQuery
+     */
+    public function getPunchoutCatalogTransactionPropelQuery(): PgwPunchoutCatalogTransactionQuery
+    {
+        return $this->getProvidedDependency(PunchoutCatalogsDependencyProvider::PROPEL_QUERY_PUNCHOUT_CATALOG_TRANSACTION);
+    }
+
+    /**
+     * @return \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionSetupQuery
+     */
+    public function getPunchoutCatalogConnectionSetupPropelQuery(): PgwPunchoutCatalogConnectionSetupQuery
+    {
+        return $this->getProvidedDependency(PunchoutCatalogsDependencyProvider::PROPEL_QUERY_PUNCHOUT_CATALOG_CONNECTION_SETUP);
+    }
+
+    /**
+     * @return \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionCartQuery
+     */
+    public function getPunchoutCatalogConnectionCartPropelQuery(): PgwPunchoutCatalogConnectionCartQuery
```

Comment:
Where is it used?

Reply:
In the methods `PunchoutCatalogsEntityManager::updatePunchoutCatalogConnectionSetup`

and `PunchoutCatalogsEntityManager::createPunchoutCatalogConnectionSetup`